services/worker-manager/src/providers/aws.js

```diff
@@ -323,6 +323,17 @@ class AwsProvider extends Provider {
         switch (is.InstanceState.Name) {
           case 'pending':
           case 'running':
+            // If the worker will be expired soon but it still exists,
+            // update it to stick around a while longer. If this doesn't happen,
+            // long-lived instances become orphaned from the provider. We don't update
+            // this on every loop just to avoid the extra work when not needed
+            if (worker.expires < taskcluster.fromNow('1 day')) {
+              await worker.update(this.db, worker => {
+                worker.expires = taskcluster.fromNow('1 week');
+              });
+            }
+            this.seen[worker.workerPoolId] += worker.capacity || 1;
```

Comment:
that line is there because of the `break` statement before the next case, prior to adding this logic for `pending` and `running` instances there was a fallthrough that includes that line.

Also, that commit is just there to show what it would look like to just do the AWS part of this and won't be landed.